osu.Game/Screens/Play/Player.cs

```diff
@@ -345,6 +345,16 @@ public override void OnEntering(IScreen last)
                 .Delay(250)
                 .FadeIn(250);
 
+            ShowStoryboard.ValueChanged += _ =>
+            {
+                if (ShowStoryboard.Value && storyboard == null)
+                    initializeStoryboard(true);
+            };
+
+            Background.EnableUserDim.Value = true;
+            storyboardContainer.StoryboardReplacesBackground.Value = Beatmap.Value.Storyboard.ReplacesBackground && Beatmap.Value.Storyboard.HasDrawable;
+            storyboardContainer.StoryboardReplacesBackground.BindTo(Background?.StoryboardReplacesBackground);
```

Comment:
this is weird binding logic. make a local private bendable rather than binding two other classes' bindables